src/render/cmd.rs

```diff
@@ -157,22 +148,25 @@ fn do_balance(balance: &mut Vec<Kind>, chars: &[char]) -> bool {
                 balance.pop();
                 true
             }
-            (Kind::Asterisk, _)
-            | (Kind::Asterisk2, _)
-            | (Kind::SingleQuota, _)
-            | (Kind::DoubleQuota, _)
-            | (Kind::Tilde, _)
-            | (Kind::Tilde2, _)
-            | (Kind::Backtick, _)
-            | (Kind::Backtick3, _) => {
+            (
+                Kind::ParentheseStart
+                | Kind::BracketStart
+                | Kind::Asterisk
+                | Kind::Asterisk2
+                | Kind::SingleQuota
+                | Kind::DoubleQuota
+                | Kind::Tilde
+                | Kind::Tilde2
+                | Kind::Backtick
+                | Kind::Backtick3,
+                _,
+            ) => {
                 balance.push(kind);
                 true
             }
             _ => false,
         }
```

Comment:
These should be equivalent, but one is obviously more readable.

Double check me on the equivalence, though, maybe I missed something.